src/ui/foundation/types.ts

```diff
@@ -0,0 +1,23 @@
+import { Command } from './SettingsProvider/types';
+
+export class CommandEventTarget extends EventTarget {
+  commands: Map<string, Command> = new Map();
+
+  public registerCommand(command: Command): void {
+    if (command.callback) {
+      this.commands.set(command.id, command);
+    }
+  }
+
+  public execute(commandId: string): void {
+    this.executeCommand(commandId);
+    this.dispatchEvent(new CustomEvent(commandId));
```

Comment:
We still need to send the event for this command even if there is no callback associated to it. Eg. `command_palette` doesn't have any callback, but the component needs to know when the hotkey is executed.

We can revisit later when CommandPalette is moved to some kind of plugin system.

Reply:
I think we need to remove callbacks of commands as they exist in our codebase.

The commands that are provided by settingsprovider are maybe something like a CommandKeyBindings.

They *bind* a *command* to a *hotkey*, but they shouldn't have a callback.

The settings provided by `SettingsProvider` should be serializable.

So basicly I would limit settingsprovider to providing an interface to the json config file, which can't contain function pointers.

Reply:
I did consider doing another interface for commands now, but it is not clear to me how callbacks will be managed in the future once we have plugins and maybe some commands triggering an event for the backend. I decided to not try to model something I don't properly understand yet.